### PR TITLE
0160-feat-merkle-nft-and-id

### LINKED ISSUE #160: State
**Description:**
Implement MerkleNFT as `FCValue` and `MerkleNftId` as `FCKey`

`MerkleNFT` should have:
- `owner`
- `position`
- `memo`
- `creationTime`
as per the design document

`MerkleNftId` should have:
- `tokenId`
- `serialNumber`

### PR DESCRIPTION
**Pull request by StoyanDenev**
**Related issue(s)**:
Closes #160 

**Summary of the change**:
MerkleNft and MerkleNftId added to the state

**External impacts**:
None.

**Applicable documentation**

### CODE REVIEW

**Comment by Daniel-K-Ivanov** [comment]
LGTM. Lets add the missing unit tests

File: hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleNft.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import com.google.common.base.MoreObjects;
+import com.hedera.services.state.serdes.DomainSerdes;
+import com.hedera.services.state.submerkle.EntityId;
+import com.hedera.services.state.submerkle.RichInstant;
+import com.swirlds.common.FCMElement;
```
Comment (failfmi) [inline hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleNft.java:27]: nit: unused line

File: hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleNft.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+import com.swirlds.common.FCMValue;
+import com.swirlds.common.io.SerializableDataInputStream;
+import com.swirlds.common.io.SerializableDataOutputStream;
+import com.swirlds.common.merkle.utility.AbstractMerkleLeaf;
+
+import javax.annotation.Nullable;
```
Comment (failfmi) [inline hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleNft.java:33]: nit: unused line

File: hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleNftId.java [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+	@Override
+	public long getClassId() {
+		return RUNTIME_CONSTRUCTABLE_ID;
+	}
+
+	@Override
+	public int getVersion() {
+		return MERKLE_VERSION;
+	}
```
Comment (failfmi) [inline hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleNftId.java:106]: Can we add tests asserting these getters? (ref. `MerkleTokenTest` -> `merkleMethodsWork`)

File: hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleNft.java [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+	@Override
+	public long getClassId() {
+		return RUNTIME_CONSTRUCTABLE_ID;
+	}
+
+	@Override
+	public int getVersion() {
+		return MERKLE_VERSION;
+	}
+
```
Comment (failfmi) [inline hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleNft.java:110]: Can we add tests asserting these getters? (ref. MerkleTokenTest -> merkleMethodsWork)

**Review by failfmi — APPROVED** [review]

**Review by Daniel-K-Ivanov — APPROVED** [review]

**Pull request merged**